url9ecf1e8297_nhamidi_ninja_tgz-pJ8-net_fec_openrq_test_TestRunnerJ8/src/RecepteurThreadMulti.java

```diff
@@ -0,0 +1,176 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.InetAddress;
+import java.net.MulticastSocket;
+
+public class RecepteurThreadMulti extends Thread {
+    
+    private int port;
+    private MulticastSocket socketReception;
+    private boolean endLoop;
+    private boolean historic;
+    int packetData;
+    private int numberOfPackets;
+    private int endReception;
+    private int nbOfReceiver;
+    private int totalSymb;
+    private long time_1;
+    private long time_2;
+    private int nbAck;
+    private long before;
+    private boolean needMore;
+    // private double redondancy;
+    private float[] tab_lost = new float[50];
+    
+    public RecepteurThreadMulti(int port, InetAddress groupeIP, int nbOfRecv, boolean historic) throws IOException {
+	this.port = port;
+	this.endLoop = true;
+	this.needMore = false;
+	this.socketReception = new MulticastSocket(port);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *socketReception*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).